language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -263,6 +263,11 @@ impl LibraVM {
             account_config::from_currency_code_string(txn.gas_currency_code())
                 .map_err(|_| VMStatus::Error(StatusCode::INVALID_GAS_SPECIFIER))
         );
+        // Check that the gas currency is valid. It is not used directly here but the
+        // results should be consistent with the checks performed for validation.
+        if let Err(err) = get_currency_info(&account_currency_symbol, remote_cache) {
+            return discard_error_vm_status(err);
```

Comment:
actually I am confused by this now... I checked but not clear and I figured I'd ask which is the easiest/best thing.
Is this going to discard the transaction? I believe we should charge the user for those mistakes, is this what we are doing?

Reply:
Yes, this will discard the transaction. The prologue is always run without gas metering so no gas cost has been incurred at this point. This also matches what we have currently written in the VM specification: at the beginning of the execution phase, transactions are revalidated and discarded if they are invalid. The issue here was that this one validation check was missing when running validation from the execution phase. (Remember my earlier comment that it would be good to share the same implementation? 😉)

Reply:
yeah I figured that was adding the missing one, I just find it strange we are not charging for those kind of errors. In fact I think we should.
But I guess it's all good, we can change it at any time and we are reasonably protected by the fact validation would discard those...
anyway, still accepted, so feel free to go in at any time

Reply:
FWIW, you can't charge for this specific error because the gas currency is invalid. More generally, in response to your comment about "validation would discard those", this *is* validation. The first step of the execution phase is to re-run validation. Except for sequence number changes, the results should always be the same as during the validation phase, so I wanted to make this change to keep that consistency. I can't think of any way you could hit this in practice.